ddtrace/tracer/writer.go

```diff
@@ -66,9 +69,13 @@ func newAgentTraceWriter(c *config, s *prioritySampler, statsdClient globalinter
 }
 
 func (h *agentTraceWriter) add(trace []*Span) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
```

Comment:
@kakkoyun I think this locking logic should be in `payload`. I wonder too if we should "finish" payloads to avoid adding traces to it.

There are tests timing out.